syntax = "proto3";
import "google/protobuf/empty.proto";

package suggestions;

service SuggestionsService {
    rpc InitSuggestions(SuggestionsData) returns (google.protobuf.Empty);
    rpc Suggest(SuggestionsRequest) returns (SuggestionsResponse);
}

message Book {
    string bookId = 1;
    string title = 2;
    string author = 3;
}

message SuggestionsData {
    string orderId = 1;
    repeated Book data = 2;
}

message SuggestionsRequest {
    string orderId = 1;
    repeated int32 vector_clock = 2;
}

message SuggestionsResponse {
    repeated int32 vector_clock = 1;
    repeated Book suggestedBooks = 2;
}
